test/e2e/configuration_test.go

```diff
@@ -2,118 +2,138 @@ package e2e_test
-				cluster.ProviderSettings.ProviderName,
-			).Should(Equal(string(userSpec.clusters[0].Spec.ProviderSettings.ProviderName)))
-			Expect(
-				cluster.ProviderSettings.RegionName,
-			).Should(Equal(userSpec.clusters[0].Spec.ProviderSettings.RegionName))
-		})
-
-		By("Update cluster\n", func() {
-			userSpec.clusters[0].Spec.ProviderSettings.InstanceSizeName = "M20"
-			utils.SaveToFile(
-				userSpec.clusters[0].ClusterFileName(),
-				utils.JSONToYAMLConvert(userSpec.clusters[0]),
-			)
-			kube.Apply(userSpec.clusters[0].ClusterFileName(), "-n", userSpec.namespace)
-		})
-
-		By("Wait cluster updating", func() {
-			waitCluster(userSpec, "2")
-		})
-
-		By("Check attributes", func() {
-			uCluster := mongocli.GetClustersInfo(userSpec.projectID, userSpec.clusters[0].Spec.Name)
-			Expect(
-				uCluster.ProviderSettings.InstanceSizeName,
-			).Should(Equal(
-				userSpec.clusters[0].Spec.ProviderSettings.InstanceSizeName,
-			), "Instance size should be the same as requested by the user")
-		})
-
-		By("Delete cluster", func() {
-			kube.Delete(userSpec.clusters[0].ClusterFileName(), "-n", userSpec.namespace)
-			Eventually(
-				checkIfClusterExist(userSpec),
-				"10m", "1m",
-			).Should(BeFalse(), "Cluster should be deleted from Atlas")
-		})
-
-		By("Delete project", func() {
-			kube.Delete(userSpec.projectPath, "-n", userSpec.namespace)
-			Eventually(
-				func() bool {
-					return mongocli.IsProjectInfoExist(userSpec.projectID)
-				},
-				"5m", "20s",
-			).Should(BeFalse(), "Project should be deleted from Atlas")
-		})
-	})
+	newData := func(description, path string) (string, testDataProvider) {
+		var data testDataProvider
+		data.description = description
+		data.confPath = path
+		data.userSpec = NewUserInputs("my-atlas-key")
+		return data.description, data
+	}
+
+	DescribeTable("Namespaced operators working only with its own namespace with different configuration",
+		func(test testDataProvider) {
+			data = test
+			mainCycle(test.confPath, test.userSpec)
+		},
+		Entry(newData("Simple Test with default cluster", "data/atlascluster_basic.yaml")),
+		Entry(newData("Simple Backup Configuration and 2 users", "data/atlascluster_backup.yaml")),
+		// Entry(newData("Multiregion, Backup and 2 users", "data/atlascluster_multiregion.yaml")), // TODO CLOUDP-83419
+	)
 })
+
+func mainCycle(clusterConfigurationFile string, userSpec userInputs) {
+	By("Prepare namespaces and project configuration", func() {
+		kube.CreateNamespace(userSpec.namespace)
+
+		project := utils.NewProject().
+			ProjectName(userSpec.projectName).
+			SecretRef(userSpec.keyName).
+			CompleteK8sConfig(userSpec.k8sProjectName)
+		utils.SaveToFile(FilePathTo(userSpec.projectName), project)
+
+		userSpec.clusters = append(userSpec.clusters, utils.LoadUserClusterConfig(clusterConfigurationFile))
+		userSpec.clusters[0].Spec.Project.Name = userSpec.k8sProjectName
+		userSpec.clusters[0].Spec.ProviderSettings.InstanceSizeName = "M10"
+		utils.SaveToFile(
```

Comment:
am I right that the thing we are doing here: load yaml file for cluster, change a couple of fields and save back?
Is it possible to specify these 2 fields directly in the file to avoid overwriting it?

Reply:
yes, you're right.

Reply:
yes, you are right, we cannot use the hardcoded project name 👍